src/jupyterscad/_openscad.py

```diff
@@ -0,0 +1,85 @@
+"""
+Jupyter SCAD
+Copyright (C) 2023 Jennifer Reiber Kyle
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later
+version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+"""
+import logging
+import platform
+import subprocess
+from pathlib import Path
+from shutil import which
+
+from .exceptions import OpenSCADException
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_OPENSCAD_EXECUTABLE = {
+    "Linux": ["/usr/bin/openscad", "/usr/local/bin/openscad"],
+    "Darwin": ["/Applications/OpenSCAD.app/Contents/MacOS/OpenSCAD"],
+}
+
+
+def process(scad_file, output_file, executable: Path = None):
+    """Generate stl from scad using OpenSCAD executable"""
+    if executable and not Path(executable).is_file():
```

Comment:
This could also go into the or'ed `which` statement:
```python
    executable = which(executable) or \
                     which("openscad") or \
                     which("OpenSCAD") or \
                     which(macPath)
```

;)